modules/fedimint-lnv2-client/src/lib.rs

```diff
@@ -99,21 +102,21 @@ impl SendOperationMeta {
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ReceiveOperationMeta {
-    pub gateway: SafeUrl,
+    pub gateway: Option<SafeUrl>,
     pub contract: IncomingContract,
-    pub invoice: LightningInvoice,
+    pub invoice: Option<LightningInvoice>,
     pub custom_meta: Value,
 }
 
 impl ReceiveOperationMeta {
     /// Calculate the absolute fee paid to the gateway on success.
-    pub fn gateway_fee(&self) -> Amount {
-        match &self.invoice {
+    pub fn gateway_fee(&self) -> Option<Amount> {
```

Comment:
So this is no longer possible when the invoice is generated using `recurringd`? How should clients calculate fees now?